compiler/rustc_middle/src/ty/context.rs

```diff
@@ -596,8 +596,16 @@ impl<'tcx> Interner for TyCtxt<'tcx> {
         self.coroutine_is_async_gen(coroutine_def_id)
     }
 
-    fn layout_is_pointer_like(self, param_env: ty::ParamEnv<'tcx>, ty: Ty<'tcx>) -> bool {
-        self.layout_of(self.erase_regions(param_env.and(ty)))
+    // We don't use `TypingEnv` here as it's only defined in `rustc_middle` and
```

Comment:
? why wouldn't `rustc_next_trait_solver` want to know about `TypingEnv`, at least not eventually?

Reply:
Because the new solver should imo never use anything that takes a `TypingEnv`. It has to handle opaques in their defining scope, and having some code paths while are only used while not defining opaque types does not feel like something we'd ever want or need :thinking: maybe when normalizing opaques with reveal-all or whatever, but we currently don't need it there and I don't expect that to change